src/Microsoft.AspNet.Mvc.Core/MvcOptions.cs

```diff
@@ -22,16 +22,16 @@ public class MvcOptions
         public MvcOptions()
         {
             Conventions = new List<IApplicationModelConvention>();
-            ModelBinders = new List<ModelBinderDescriptor>();
+            ModelBinders = new List<IModelBinder>();
             ViewEngines = new List<ViewEngineDescriptor>();
```

Comment:
@davidfowl, that's certainly a possibility. There are types like the `IMvcRazorHost` that are created as part of the `RazorViewEngine`'s object graph that are intended to be transient. We'd have to tweak those to ensure they're ok with longer lifespans or change how we consume them.